### PR TITLE
Example for conanrun

### PR DESCRIPTION
**Pull request by franramirez688**
Docs for https://github.com/conan-io/docs/issues/2465

### CODE REVIEW

File: reference/conanfile/tools/env/virtualrunenv.rst [hunk 1]
```diff
@@ -60,6 +60,43 @@ After the execution of one of those files, a new deactivation script will be gen
...
+
+    import os
+    from conan import ConanFile
+
+    class HelloConan(ConanFile)
+
+        # ...
+
+        def package_info(self):
+            self.runenv_info.append("PYTHONPATH", os.path.join(self.package_folder, "my-site-packages"))
```
Comment (memsharded) [inline reference/conanfile/tools/env/virtualrunenv.rst:78]:
```suggestion
            self.runenv_info.append_path("PYTHONPATH", os.path.join(self.package_folder, "my-site-packages"))
```
Maybe even prepend_path?

File: reference/conanfile/tools/env/virtualrunenv.rst [hunk 2]
```diff
@@ -60,6 +60,43 @@ After the execution of one of those files, a new deactivation script will be gen
...
+
+    class HelloTestConan(ConanFile):
+        # VirtualBuildEnv and VirtualRunEnv can be avoided if "tools.env.virtualenv:auto_use" is defined
+        # (it will be defined in Conan 2.0)
+        generators = "VirtualRunEnv"
+
+        # ...
+
+        def test(self):
+            self.run("set PYTHONPATH", env="conanrun")
```
Comment (memsharded) [inline reference/conanfile/tools/env/virtualrunenv.rst:94]: This is Windows only, better clarify or remove it

**Review by memsharded — COMMENTED** [review]
Is this extracted from some running test? Maybe it is worth copy & pasting from a test to avoid possible minor issues?

File: reference/conanfile/tools/env/virtualrunenv.rst (hunk 2 above)
Comment (franramirez688) [reply reference/conanfile/tools/env/virtualrunenv.rst:94]: Oh, sure!

File: reference/conanfile/tools/env/virtualrunenv.rst (hunk 1 above)
Comment (franramirez688) [reply reference/conanfile/tools/env/virtualrunenv.rst:78]: I can tell it in a comment

Comment (memsharded) [reply reference/conanfile/tools/env/virtualrunenv.rst:78]: In any case, the ``_path`` is necessary, isn't it?

Comment (franramirez688) [reply reference/conanfile/tools/env/virtualrunenv.rst]: that's it

**Comment by franramirez688** [comment]
> Do we need the entire source code? Maybe just use the relevant parts?

Perhaps, it's clearer to see a full-working example. I guess it's fair enough to put the relevant parts but a few lines more are worth it... 

> Also, shouldn't we document this in the [conanfile.py](https://docs.conan.io/en/1.3/reference/conanfile/other.html) section? About the env argument in the run method and that it's defaulted to conanbuild ?

I already added some info there. Have a look at this PR: https://github.com/conan-io/docs/pull/2516/files If you think it's not enough I can add something more.

File: reference/conanfile/tools/env/virtualrunenv.rst [hunk 3]
```diff
@@ -60,6 +60,45 @@ After the execution of one of those files, a new deactivation script will be gen
 environment, so the environment can be restored when desired. The file will be named also following the
 current active configuration, like ``deactivate_conanrunenv-release-x86_64.bat``.
 
+Given that, let's see, for instance, how to set a ``MYPATH`` variable through the ``runenv_info`` to be applied when running a command
+from ``test()`` side:
```
Comment (czoido) [inline reference/conanfile/tools/env/virtualrunenv.rst:64]: Maybe we can rephrase, something similar to this?

```suggestion
Let's see and example on how to add an environment variable to the runenv_info and getting its value later in the consumer explicitly invoking the `conanrun` environment.
```

**Review by memsharded — APPROVED** [review]

**Pull request merged**